### PR TITLE
Fixes after tests from Validation plan

### PR DESCRIPTION
**Pull request by alainbodiguel**

### CODE REVIEW

File: subscriptions-manager/src/main/java/io/arlas/subscriptions/service/UserSubscriptionManagerService.java [hunk 1]
```diff
@@ -73,24 +69,22 @@ public UserSubscription postUserSubscription(UserSubscription userSubscription,
...
-        logger.debug(String.format("User %s requests subscription %s (deleted %b)", user.orElse(UNKNOWN_USER), id, deleted));
         return this.daoDatabase.getSubscription(id, user, deleted);
     }
 
     public void deleteUserSubscription(UserSubscription userSubscription) throws ArlasSubscriptionsException {
-        logger.debug(String.format("User %s deletes subscription %s", userSubscription.created_by, userSubscription.getId()));
         this.daoDatabase.setUserSubscriptionDeletedFlag(userSubscription, true);
 
         try {
             this.daoIndexDatabase.setUserSubscriptionDeletedFlag(userSubscription, true);
```
Comment (laurent-thiebaud-gisaia) [inline subscriptions-manager/src/main/java/io/arlas/subscriptions/service/UserSubscriptionManagerService.java]: Doesn't a setter return an updated instance of the UserSubscription (it is strange to update it manually)?

File: subscriptions-matcher/src/main/java/io/arlas/subscriptions/service/SubscriptionsService.java [hunk 2]
```diff
@@ -66,7 +65,7 @@
                     result.addAll(items.getHits());
                     next = items.getLinks() != null ? items.getLinks().get("next") : null;
                     if (next != null) {
-                        queryParams = getQueryParams(next.getHref().split("\\?")[1]);
+                        queryParams = getQueryParams(next.getHref().split("\\?")[1].replace("before=&", ""));
```
Comment (laurent-thiebaud-gisaia) [inline subscriptions-matcher/src/main/java/io/arlas/subscriptions/service/SubscriptionsService.java]: Where does the `before` come from?

Comment (alainbodiguel) [reply subscriptions-matcher/src/main/java/io/arlas/subscriptions/service/SubscriptionsService.java]: It's a bug from ARLAS server. I replace it as a workaround until a patched version is released.

File: subscriptions-manager/src/main/java/io/arlas/subscriptions/service/UserSubscriptionManagerService.java (hunk 1 above)
Comment (alainbodiguel) [reply subscriptions-manager/src/main/java/io/arlas/subscriptions/service/UserSubscriptionManagerService.java]: Moving it to dao as for the other updates.

File: subscriptions-manager/src/main/java/io/arlas/subscriptions/dao/MongoUserSubscriptionDAOImpl.java [hunk 3]
```diff
@@ -143,6 +143,7 @@ private UserSubscription convertDocument(Document document) {
     @Override
     public void setUserSubscriptionDeletedFlag(UserSubscription userSubscription, boolean isDeleted) throws ArlasSubscriptionsException {
         try {
+            userSubscription.setDeleted(true);
```
[the remark this answers is not in the document]
Comment (alainbodiguel) [reply subscriptions-manager/src/main/java/io/arlas/subscriptions/dao/MongoUserSubscriptionDAOImpl.java]: Good point!

**Pull request merged**